### PR TITLE
DPC-1265, DPC-1296, DPC-1314: Provider and vendor ids are only used in prod-sbx

### PR DESCRIPTION
**Pull request by Sun-Mountain**
**Why**

Update deployment so that in each env we're setting the env variable - from there, the native Rails ability to know what env we're in will work. Before we change AWS env, we'll need to know what configs we want set up (what we call them, right settings, etc.)

To confirm, the environment conditional, it was combined with DPC-1296 to only create `provider_id` and `vendor_id` in prod-sbx only with some minor view conditionals.

**What Changed**

In the organization model, the method `prod_sbx?` checks if `DEPLOY_ENV` is set to `prod-sbx`.

If `DEPLOY_ENV == prod-sbx`:
- `vendor_id` will be assigned to Health IT Vendors and be used in place of an NPI
- `provider_id` will be assigned to provider organizations and be used if no NPI provided
- `provider_id` or `vendor_id` can be used in place of an NPI in the API

In all other environments:
- NPI is required for an organization to be enabled in the NPI

**Choices Made**

In other environments:
Screen Shot 2020-04-21 at 1 18 59 PM
Screen Shot 2020-04-21 at 1 19 11 PM
Screen Shot 2020-04-21 at 2 02 32 PM
If no NPI is present:
Screen Shot 2020-04-21 at 2 02 45 PM

In `prod-sbx`:
Screen Shot 2020-04-21 at 1 31 46 PM
Screen Shot 2020-04-21 at 4 48 44 PM
Screen Shot 2020-04-21 at 4 48 59 PM

**Tickets closed**:

[DPC-1265](https://jiraent.cms.gov/browse/DPC-1265)
[DPC-1296](https://jiraent.cms.gov/browse/DPC-1296)
[DPC-1314](https://jiraent.cms.gov/browse/DPC-1314)

**Future Work**

[DPC-1295](https://jiraent.cms.gov/browse/DPC-1295)

**Checklist**

- [x] All tests are passing via `make ci-app` (app change) and `make ci-web` (website change)
- [ ] Swagger documentation has been updated
- [ ] FHIR documentation has been updated
- [ ] Any required dpc-ops changes have a PR submitted and mentioned in this ticket
- [ ] Any manual migration steps are documented, scripts written (where applicable), and tested
- [ ] Before merging, any required dpc-ops changes have been approved and merged into master of the dpc-ops repo

### CODE REVIEW

File: dpc-web/app/helpers/aws_env_helper.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module AwsEnvHelper
+  def prod_sbx?
+    ENV['AWS_ENV'] == 'prod-sbx'
```
Comment (dhgreene) [inline dpc-web/app/helpers/aws_env_helper.rb:5]: We're using `DEPLOY_ENV` instead, right?

File: dpc-web/spec/helpers/aws_env_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe AwsEnvHelper, type: :helper do
+  describe '#aws_environment' do
+    context 'successful AWS_ENV check' do
+      it 'returns true if AWS_ENV equals prod-sbx' do
+        allow(ENV).to receive(:[]).with('AWS_ENV').and_return('prod-sbx')
```
Comment (dhgreene) [inline dpc-web/spec/helpers/aws_env_helper_spec.rb:9]: If we're no longer using `AWS_ENV` a few things in this spec will need to change.

File: dpc-web/app/helpers/aws_env_helper.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module AwsEnvHelper
```
Comment (dhgreene) [inline dpc-web/app/helpers/aws_env_helper.rb:3]: Now that you've added `organization.prod_sbx?` do you need this helper?

Comment (Sun-Mountain) [reply dpc-web/app/helpers/aws_env_helper.rb:3]: No, I'm deleting this helper. I just wanted to run some code climate tests.

**Pull request closed without merging**